### PR TITLE
This fix #1647. Checking for voucher availability is working correctly

### LINKED ISSUE #1647: Offer condition in Applicator.get_basket_offers always returns True.
`Applicator.get_basket_offers()` checks for applicable basket offers using something like `if voucher.is_available_to_user(user)` (see https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/offer/utils.py#L99). However, the return value of `is_available_to_user()` is a 2-tuple, so this will always evaluate to True.

### PR DESCRIPTION
**Pull request by jonasborges**
Checking for voucher availability is working correctly on Applicator.get_basket_offers(). 
Tests updated.

### CODE REVIEW

File: src/oscar/apps/offer/utils.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def get_basket_offers(self, basket, user):
             return offers
 
         for voucher in basket.vouchers.all():
-            if voucher.is_active() and voucher.is_available_to_user(user):
```
Comment (codeinthehole) [inline src/oscar/apps/offer/utils.py]: I'm not a fan of boolean variables like `no_message_on_result` - I would rather we used an index in the below conditional or read the two return values into variable - perhaps this:
```python
if voucher.is_active() and voucher.is_available_to_user(user=user)[0]:
```

**Comment by codeinthehole** [comment]
Thanks for the PR - let me know what you think about my suggestion

**Comment by jonasborges** [comment]
Hello, David!

I agree, maybe <code>no_message_on_result</code> wasn't a good idea.
However, I prefer to treat the result before the test condition. Like this:

```python 
is_voucher_available_to_user, _ = voucher.is_available_to_user(user=user)
if voucher.is_active() and is_voucher_available_to_user:
```

I think it's more clear to read. Don't you?

**Comment by codeinthehole** [comment]
@jonasborges That's fine by me - make those changes and I'll merge it.

PS - it's slightly better to use `__` rather than `_` for throw-away variables. `_` is normally an alias for the gettext translation function so it's better to use something different.

**Comment by jonasborges** [comment]
@codeinthehole It's done.

I'm new on that, so thanks for the tip! :D

**Comment by maikhoepfel** [comment]
Thank you. I cherry-picked your changes across in 1d650a1

**Pull request closed without merging**